rwengine/src/ai/CharacterController.cpp

```diff
@@ -239,6 +239,28 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
         } else if (character->animator->getAnimation(AnimIndexAction) ==
                    anm_enter) {
             if (character->animator->isCompleted(AnimIndexAction)) {
+                if (vehicle->getVehicle()->vehiclename_ == "TAXI"
+                    || vehicle->getVehicle()->vehiclename_ == "CABBIE"
+                    || vehicle->getVehicle()->vehiclename_ == "BORGNINE") {
```

Comment:
Correct, this could be done in either of those as well, however comparing to the original behavior, both of them would result in character receiving the perk too early.
It goes like this:
```
I [World] Creating Vehicle ID 119 (BANSHEE)
CharacterObject.cpp:420
CharacterObject.cpp:420
VehicleObject.cpp:519
CharacterController.cpp:242
```
The :242 line (where the code currently resides) is being called once the player's inside a vehicle, not once the process of entering a vehicle has been initiated. I guess it should be fairly easy to move it, and I could move it right now wherever you feel like it fits but it won't be as close to what happens in vanilla as the current placement.